src/module_wrap.cc

```diff
@@ -142,6 +142,57 @@ v8::Maybe<bool> ModuleWrap::CheckUnsettledTopLevelAwait() {
   return v8::Just(false);
 }
 
+static Local<Object> createImportAttributesContainer(
+    Realm* realm,
+    Isolate* isolate,
+    Local<FixedArray> raw_attributes,
+    const int elements_per_attribute) {
+  CHECK_EQ(raw_attributes->Length() % elements_per_attribute, 0);
+  Local<Object> attributes =
+      Object::New(isolate, v8::Null(isolate), nullptr, nullptr, 0);
+  for (int i = 0; i < raw_attributes->Length(); i += elements_per_attribute) {
+    attributes
+        ->Set(realm->context(),
+              raw_attributes->Get(realm->context(), i).As<String>(),
+              raw_attributes->Get(realm->context(), i + 1).As<Value>())
+        .ToChecked();
+  }
+
+  return attributes;
+}
+
+static Local<Array> createModuleRequestsContainer(
+    Realm* realm, Isolate* isolate, Local<FixedArray> raw_requests) {
+  Local<Array> requests = Array::New(isolate, raw_requests->Length());
+  for (int i = 0; i < raw_requests->Length(); i++) {
+    Local<ModuleRequest> module_request =
+        raw_requests->Get(realm->context(), i).As<ModuleRequest>();
+
+    Local<String> specifier = module_request->GetSpecifier();
+
+    Local<FixedArray> raw_attributes = module_request->GetImportAssertions();
+    Local<Object> attributes =
+        createImportAttributesContainer(realm, isolate, raw_attributes, 3);
+
+    Local<v8::Name> names[] = {
+        realm->isolate_data()->specifier_string(),
+        realm->isolate_data()->attributes_string(),
+    };
+    Local<Value> values[] = {
+        specifier,
+        attributes,
+    };
+    DCHECK_EQ(arraysize(names), arraysize(values));
+
+    Local<Object> request = Object::New(
+        isolate, v8::Null(isolate), names, values, arraysize(names));
+
+    requests->Set(realm->context(), i, request).ToChecked();
```

Comment:
You can collect the requests into a std::vector and them pass them into `Array::New`, that will be faster than doing `arr->Set` repeatedly.